File: works/11053.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    int answer = 1;
    vector<int> v;
    int n; cin >> n;
    for(int i = 0; i < n; i++)
    {
        int num; cin >> num;
        v.push_back(num);
    }

    int d[1010] = {0,};
    d[0] = 1;
    for(int i = 1; i < n; i++)
    {
        int highest = 0;
        for(int j = 0; j < i; j++)
        {
            if(v[j] < v[i])
            {
                highest = max(highest, d[j]);
            }
        }
        d[i] = highest + 1;
        answer = max(answer, d[i]);
    }

    cout << answer;
}

File: works/11053.kt
import java.util.*
import java.io.*;
import java.io.BufferedWriter;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

fun main()
{
    val sc = Scanner(System.`in`)
    val br = BufferedReader(InputStreamReader(System.`in`))

    var n = br.readLine().toInt()
    var v = br.readLine().split(' ').map{it.toInt()}

    var answer = 1
    var d = Array<Int>(1010){0}
    d[0] = 1
    for(i in 1 until n)
    {
        var highest = 0
        for(j in 0 until i)
        {
            if(v[j] < v[i]) highest = Math.max(highest, d[j])
        }
        d[i] = highest + 1
        answer = Math.max(answer, d[i])
    }
    print(answer)
}